### PR TITLE
feat(ML-78): add script for getting incorrectly labeled assets using cleanlab

### PR DESCRIPTION
**Pull request by gheorghetutunaru**

### CODE REVIEW

File: utils/helpers.py [hunk 1]
```diff
@@ -158,7 +158,7 @@ def get_assets(
             "labels": [
                 line
                 for line in sorted(a["labels"], key=lambda l: l["createdAt"])
-                if line["labelType"] in label_types
+                if (line["labelType"] in label_types and line["jsonResponse"])
```
Comment (gheorghetutunaru) [inline utils/helpers.py:161]: Added this check since if the last label was skipped, it would return an asset with no jsonResponse

File: label_errors.py [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+                data_dir=data_path,
+                model_dir=model_path,
+                model_name=model_name,
+                training_epochs=training_epochs,
+            )
+
+            print()
+            kili_print("Number of wrong labels found: ", len(found_errors))
+
+            if found_errors:
```
Comment (PierreLeveau) [inline label_errors.py:155]: add the code below into a `upload_errors` method in `utils.helpers`

File: utils/cleanlab/train_cleanlab.py [hunk 3]
```diff
@@ -0,0 +1,271 @@
...
+from torch.optim import lr_scheduler
+import torch.backends.cudnn as cudnn
+from torchvision import datasets, models, transforms
+
+from utils.constants import ModelName
+
+
+def train_model(
+    model, criterion, optimizer, scheduler, dataloaders, device, dataset_sizes, num_epochs=10
+):
```
Comment (PierreLeveau) [inline utils/cleanlab/train_cleanlab.py:20]: add `verbose` int parameter, and print only if verbose >= 1

File: utils/cleanlab/train_cleanlab.py [hunk 4]
```diff
@@ -0,0 +1,271 @@
...
+    """
+    Method that trains the given model and return the best one found in the given epochs
+    """
+    since = time.time()
+
+    best_model_wts = copy.deepcopy(model.state_dict())
+    best_acc = 0.0
+
+    for epoch in range(num_epochs):
+        print(f"Epoch {epoch + 1}/{num_epochs}")
```
Comment (PierreLeveau) [inline utils/cleanlab/train_cleanlab.py:30]: only print if `verbose >= 1`

File: label_errors.py [hunk 5]
```diff
@@ -0,0 +1,184 @@
...
+    default=None,
+    help="Model name (one of efficientnet_b0, resnet50)",
+)
+@click.option("--project-id", default=None, help="Kili project ID")
+@click.option(
+    "--training-epochs",
+    default=10,
+    type=int,
+    help="Number of epochs to train each CV fold",
+)
```
Comment (PierreLeveau) [inline label_errors.py:63]: add `verbose` (see below)

File: label_errors.py [hunk 6]
```diff
@@ -0,0 +1,184 @@
...
+                    time.sleep(throttling_per_call - toc)
+
+            model_name = set_default(
+                model_name,
+                ModelName.EfficientNetB0,
+                "model_name",
+                [ModelName.EfficientNetB0, ModelName.Resnet50],
+            )
+
+            found_errors = train_and_get_error_labels(
```
Comment (PierreLeveau) [inline label_errors.py:144]: add verbose parameter

**Review by PierreLeveau — CHANGES_REQUESTED** [review]

File: label_errors.py (hunk 2 above)
Comment (gheorghetutunaru) [reply label_errors.py:178]: Added two methods, one for saving the json and one for actually uploading to Kili

File: label_errors.py [hunk 7]
```diff
@@ -0,0 +1,187 @@
...
+)
+@click.option("--max-assets", default=None, type=int, help="Maximum number of assets to consider")
+@click.option(
+    "--model-name",
+    default=None,
+    help="Model name (one of efficientnet_b0, resnet50)",
+)
+@click.option("--project-id", default=None, required=True, help="Kili project ID")
+@click.option(
+    "--training-epochs",
```
Comment (PierreLeveau) [inline label_errors.py:94]: rename to `epochs`

File: label_errors.py [hunk 8]
```diff
@@ -0,0 +1,187 @@
...
+def main(
+    api_endpoint: str,
+    api_key: str,
+    cv_folds: int,
+    clear_dataset_cache: bool,
+    label_types: str,
+    max_assets: int,
+    model_name: str,
+    project_id: str,
+    training_epochs: int,
```
Comment (PierreLeveau) [inline label_errors.py:117]: `epochs`

File: label_errors.py [hunk 9]
```diff
@@ -0,0 +1,187 @@
...
+
+            print()
+            kili_print("Number of wrong labels found: ", len(found_errors))
+
+            if found_errors:
+                save_errors(found_errors, job_path)
+                if upload_errors:
+                    upload_errors_to_kili(found_errors, kili)
+        else:
+            kili_print("not implemented yet")
```
Comment (PierreLeveau) [inline label_errors.py:183]: `raise NotImplementedError`

**Review by PierreLeveau — CHANGES_REQUESTED** [review]
a few other changes

File: label_errors.py (hunk 9 above)
Comment (gheorghetutunaru) [reply label_errors.py]: I removed the `kili_print()`, not sure if I had to replace it with the `raise` or just add the `raise` after

**Review by PierreLeveau — APPROVED** [review]
